src/platform/selfsign_openssl.c

```diff
@@ -200,55 +256,113 @@ CxPlatTlsGenerateSelfSignedCert(
         goto Exit;
     }
 
-    Fd = fopen(PrivateKeyFileName, "wb");
+    if (!OutputPkcs12) {
+
+        Fd = fopen(PrivateKeyFileName, "wb");
+
+        if (Fd == NULL) {
```

Comment:
nit: generally we follow the pattern of `if` checks immediately follow the functions.
```suggestion
        Fd = fopen(PrivateKeyFileName, "wb");
        if (Fd == NULL) {
```